ethcore/src/tx_filter.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Smart contract based transaction filter.
+
+use std::sync::Weak;
+use std::collections::HashMap;
+use std::collections::hash_map::Entry;
+use native_contracts::TransactAcl as Contract;
+use client::{BlockChainClient, BlockId, ChainNotify};
+use util::{Mutex, Address, H256, Bytes, RwLock};
+use futures::{self, Future};
+use spec::CommonParams;
+use transaction::{Action, SignedTransaction};
+use hash::KECCAK_EMPTY;
+
+const MAX_CACHE_SIZE: usize = 4096;
+
+mod tx_permissions {
+    pub const _ALL: u32 = 0xffffffff;
```

Comment:
why `_ALL` and not `ALL` ?

Reply:
could use the bitflags crate for this

Reply:
I did not want to introduce another dependency just for this. This type is used internally so a simple `u32` is enough.